src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -657,14 +659,70 @@ public int onBrowseUp() {
         return moveCount;
     }
 
+    private void listDirectoryWithAnimationDown(final OCFile file) {
+        Animation fadeOutFront = AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadeout_front);
+        fadeOutFront.setAnimationListener(new Animation.AnimationListener() {
+            @Override
+            public void onAnimationStart(Animation animation) {
+
+            }
+
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                listDirectory(file/*, MainApp.getOnlyOnDevice()*/);
+                Animation fadeInBack =
+                        AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_back);
+                getListView().setAnimation(fadeInBack);
+            }
+
+            @Override
+            public void onAnimationRepeat(Animation animation) {
+
+            }
+        });
+        getListView().startAnimation(fadeOutFront);
+    }
+
+    private void listDirectoryWidthAnimationUp(final OCFile file) {
+        if(getListView().getVisibility() == View.GONE) {
+            listDirectory(file);
+            Animation fadeInFront =
+                    AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_front);
+            getListView().startAnimation(fadeInFront);
+            return;
+        }
+
+        Animation fadeOutBack = AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadeout_back);
+        fadeOutBack.setAnimationListener(new Animation.AnimationListener() {
+            @Override
+            public void onAnimationStart(Animation animation) {
+
+            }
+
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                listDirectory(file);
+                Animation fadeInFront =
+                        AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_front);
+                getListView().startAnimation(fadeInFront);
+            }
+
+            @Override
+            public void onAnimationRepeat(Animation animation) {
+
+            }
+        });
+        getListView().startAnimation(fadeOutBack);
+    }
+
     @Override
     public void onItemClick(AdapterView<?> l, View v, int position, long id) {
         OCFile file = (OCFile) mAdapter.getItem(position);
         if (file != null) {
             if (file.isFolder()) {
                 // update state and view of this fragment
                 // TODO Enable when "On Device" is recovered ?
-                listDirectory(file/*, MainApp.getOnlyOnDevice()*/);
+                listDirectoryWithAnimationDown(file  /*, MainApp.getOnlyOnDevice()*/);
```

Comment:
Do we need the second parameter `MainApp.getOnlyOnDevice()` or we can just delete it?

Reply:
I'm not sure, I don't know what MainApp.getOnlyOnDevice() should do. I't is not defined however.